automation_tools/__init__.py

```diff
@@ -1322,18 +1322,18 @@ def product_install(distribution, create_vm=False, certificate_url=None,
     if distribution.startswith('satellite6'):
         execute(setup_default_docker, host=host)
         execute(katello_service, 'restart', host=host)
+        if os.environ.get('LIBVIRT_KEY_URL') is not None:
+            execute(setup_libvirt_key, host=host)
         if not distribution.endswith('upstream'):
-            if os.environ.get('LIBVIRT_KEY_URL') is not None:
-                execute(setup_libvirt_key, host=host)
             if satellite_version != '6.0':
                 execute(install_puppet_scap_client, host=host)
             if satellite_version == '6.1':
                 execute(setup_oscap, host=host)
             if satellite_version in ('6.2', ''):
                 execute(oscap_content, host=host)
-            if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
-                execute(setup_foreman_discovery, host=host)
             execute(enable_ostree, host=host)
+        if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
```

Comment:
coz the task that is called uses that env var... ugly

we are going to refactor so that tasks will have params instead.